### ISSUE TITLE
Importing SHVC with AVC encoded base layer in MP4 box

### ISSUE THREAD

**Issue by amit89infotech** [gpac/gpac#450]
I have an SHVC encoded bitstream with an AVC encoded base layer. when I try to import this, I get an unhandled exception. I think the solution to this should be in the media_import.c file in the media tools folder of libgpac. Has this been done before ? Any leads will be really useful. 

Thanks for reporting your issue. Please make sure these boxes are checked before submitting your issue - thank you!

- [ yes] I looked for a similar issue and couldn't find any.
- [yes ] I tried with the latest version of GPAC. Installers available at http://gpac.io/downloads/gpac-nightly-builds/
- [ yes] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

Detailed guidelines: http://gpac.io/2013/07/16/how-to-file-a-bug-properly/

**Comment by jeanlf** [comment]
this is currently not supported. If you could provide a sample bitstream we may try to implement this

**Comment by amit89infotech** [comment]
Hi,

I have attached the sample bitstream. Could you please check it.

Thanks,
Amit

**Comment by amit89infotech** [comment]
Hi,

Is there any update on this ?

Thanks,
Amit

**Comment by rbouqueau** [comment]
Hi Amit, have you considered contributing to GPAC? If this feature is important for you, that may be a good starting point to contribute. At least we'd be happy to help :)

**Comment by amit89infotech** [comment]
Yes, I am working on it too. I would let you guys know if I am able to fix
this

**Comment by rbouqueau** [comment]
please ask if you need pointers on what to modify

**Comment by amit89infotech** [comment]
According to my analysis, I think there must be a condition in the
media_import.c while which checks for the base layer in SHVC. The
function gf_import_hevc is likely the one. Correct me if I am wrong. It
should read an avc coded bitstream when the base of SHVC is AVC

**Comment by nguyen-viet-thanh-trung** [comment]
@amit89infotech Could you please upload your bitstream to https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95. I didn't see your attached file

**Comment by amit89infotech** [comment]
I have uploaded it there.

Thanks

**Comment by nguyen-viet-thanh-trung** [comment]
HI,
Seem that there was a problem with your file PeopleOnStreet-2xAvc.bin. Is this an AVC encoded stream ? When I tested: MP4Box add PeopleOnStreet-2xAvc.bin:FMT=AVC -new out.mp4, MPBox returned Bitstream Not Compliant.
We need:
1. one bitstream encoded in AVC (base layer) as .bin
2.  one bitstream encoded in SHVC (enhancement layer) as .bin

**Comment by amit89infotech** [comment]
The stream I have uploaded is an SHVC encoded bit stream. However this has
been encoded using an AVC encoded base layer. This means that there has
been an AVC encoded sequence which has been reconstructed in the .yuv
format. This reconstructed .yuv sequence is used as the input to the base
layer of the SHVC encoder and the resulting bit stream is the one I have
uploaded previously. I will upload pertinent files along with the bit
streams you have requested.

P.S -  Just to make things clear, I have now uploaded the AVC encoded bit
stream as.bin. The SHVC encoded bit stream "PeopleOnStreet-2xAvc.bin" is
already uploaded. This is the command that I have used
MP4Box add PeopleOnStreet-2xAvc.bin:FMT=SHVC -new out.mp4

This is the output I get

HEVC import - frame size 1920 x 1080 at 30.000 FPS
[AVC/HEVC] Not enough bits in bitstream !!
[AVC/HEVC] Not enough bits in bitstream !!
HEVC Import results: 1 samples (304 NALUs) - Slices: 0 I 0 P 150 B - 150
SEI - 0 IDR
Saving C:\Amit\K2_SPDY\mp4BoxTest\POSavc.mp4: 0.500 secs Interleaving


Thanks,
Amit

**Comment by nguyen-viet-thanh-trung** [comment]
Could you please checkout and test from branch shm6 ? Command should be:
MP4Box -add base.264 -add enhanc.bin:FMT=SHVC -new out.mp4.

PS: While testing with your uploaded stream, I found that the enhancement stream was not aligned with the base stream (i.e  CTS of a sample in enhancement layer is different to in base layer). Could you please verify your encoder ?

**Comment by amit89infotech** [comment]
Thanks for the update.

I checked out the updated branch. This successfully imports an external
base layer. I could  play the AVC base layer content back on a windows
based Mp4 player. However the openHEVC decoder doesnt support this kind of
hybrid stream for playback.
The streams that I have uploaded previously are from the following encoders

Enhancement layer : SHM software: Encoder Version [7.0
(HM-15.0)][Windows][VS 1700][32 bit]
Base Layer: JM 19.0

I have uploaded another sample of AVC.bin and SHVC.bin. You might as well
check if  the same thing is observed.
The files I have uploaded are PeopleOnTheStreet2xAvc8.0.bin and POS, bin
and the command is

MP4Box -add  POS.bin:FMT=AVC:fps=30 -add
PeopleOnTheStreet2xAvc8.0.bin:FMT=SHVC:fps=30 -new POSavc8.0.mp4

Best Regards.
Amit
